File: Partie_3/lab8/python/var.py
def variance(tab):
    n = len(tab)
    if n == 0:
        return 0  # Éviter la division par zéro si le tableau est vide
    mean = sum(tab) / n  # Calcul de la moyenne
    variance1 = sum((x - mean) ** 2 for x in tab) / (n-1)  # Calcul de la variance
    return variance1

File: Partie_3/lab11/var.c
#include <stdio.h>
#include "var.h"

void variance(int t[], int n, float * var) {
    float sum = 0.0;
    float mean;
    
    for (int i = 0; i < n; i++) {
        sum += t[i];
    }
    
    mean = sum / n;
    
    float sum_of_squares = 0.0;
    for (int i = 0; i < n; i++) {
        sum_of_squares += (t[i] - mean) * (t[i] - mean);
    }
    
    *var = sum_of_squares / (n-1);
}